tree/ntuple/v7/doc/specifications.md

```diff
@@ -1,159 +1,558 @@
+### Locators and Envelope Links
 
-### GroupUuid
-The identifier used to identify a group of RNTuples that belong to each other (for instance, when an ntuple is written out as a result of processing another ntuple). Aside from this pluralistic trait, it possesses the same characteristics described in the field, "OwnUuid."
+A locator is a generalized way to specify a certain byte range on the storage medium.
+For disk-based storage, the locator is just byte offset and byte size.
+For object stores, the locator can specify a certain object ID.
+The locator as the following format
 
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                             Size              |     Type    |T|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                             Offset                            +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
 
-### FieldDescriptors
-These hold information about each field, which is structured as list of structs. Each struct has key-value pair types, which includes the following:
+_Size_: If type is zero, the number of bytes to read, i.e. the compressed size of the referenced block.
+Otherwise the size of the locator itself.
 
-#### FieldID
-The identifying number for the respective field. It follows the Integer type standards.
+_T(ype)_: Zero for an on-disk or in-file locator, 1 otherwise.
+Can be interpreted as the sign bit of the size, i.e. negative sizes indicate non-disk locators.
+In this case, the locator should be interpreted like a frame, i.e. size indicates the _size of the locator itself_.
+Only for non-disk locators, the last 8 bits of the size should be interpreted as a locator type.
+To determine the locator type, the absolute value of the 8bit integer should be taken.
+The type can take one of the following values
 
-#### FieldVersion
-The specific version with respect to the mapping from the RNTuple logical layer (i.e. fields) to the RNTuple primitives layer (i.e. columns). There are two numbers: the first number is the version as written, and the second number is the minimum version number that the reader has to support in order to be able to use the data.
+| Type | Meaning         |
+|------|-----------------|
+| 0x01 | 64bit object ID |
+| 0x02 | URI string      |
 
-#### TypeVersion
-The specific version with respect to the memory layout of the C++ type that corresponds to the field.
+_Offset_:
+For on-disk / in-file locators, the 64bit byte offset of the referenced byte range.
```

Comment:
Do we specify anywhere was the base of the offset (current position and/or start of the file)?